iot_config/src/route.rs

```diff
@@ -309,6 +308,13 @@ pub async fn delete_euis(
             break;
         }
     }
+
+    let id = Uuid::try_parse(id)?;
+    sqlx::query(" delete from route_eui_pairs where route_id = $1 ")
```

Comment:
I'm not sure which is the correct order here. 

If we remove EUIs from HPR _then_ delete, if the delete query fails (don't know why it would, but...) we would be in a state where routing for EUIs would stop until any HPR in a region with one of those devices has to reconnect. 

If we delete first _then_ remove from HPR, we won't be removing anything from routing that could still exist in the DB. And if the removes fail for some reason, HPRs wouldn't receive them on a reconnect.

Reply:
so treating this as a true "stream" from the database we more or less _have_ to remove from HPR before deleting them from the DB b/c if we did it in the reverse order we'd be deleting the records while there were still being streamed out of the DB; the previous impl only works b/c it's loading them all into memory at once before issuing the DB delete

however, after the changes we discussed on the HPR side where you're more durably persisting eui pairs and devaddr ranges and only adding/removing routes when they become enabled or disabled, i _think_ switching the order here is the correct way to process things in a failure scenario. since this RPC is serialized and won't return to the caller until the entire operation completes if the request fails midway during HPR eui removal but before deleting from the DB then the caller will receive the failure response and can re-request the deletion. if the caller received the `Ok` response that should signal that the full list was streamed for removal to the HPRs _and_ the db deletion completed successfully. if it were somehow possible to delete first and then stream for removal without deleting the source out from under the stream as you suggested the stream operation could fail after the DB delete succeeded and re-requesting would just return a no-op response b/c those eui pairs would already be gone from the DB and there wouldn't be anything to re-trigger the HPR removal stream to clean up routing

Reply:
We just removed the `delete all` apis which makes this a moot point